lib/arel/nodes/postgresql.rb

```diff
@@ -43,6 +46,11 @@ def visit_Arel_Nodes_ArrayConcat o, collector
           visit o.right, collector
         end
       end
+
+      def visit_Arel_Nodes_UnionDistinct o, collector
```

Comment:
This method will work only in Rails >= 4.2
To support earlier versions, I'd like to ask you to create a method for Rails < 4.2 as well, like it is done for other methods in this file